### PR TITLE
RTFM command

### PR DESCRIPTION
**Pull request by Coler6gamer**

### CODE REVIEW

File: carberretta/extensions/meta.py [hunk 1]
```diff
@@ -159,11 +161,59 @@ async def cmd_stats(ctx: lightbulb.SlashContext) -> None:
...
     if not bot.d.loc:
         bot.d.loc = CodeCounter().count()
     bot.add_plugin(plugin)
+    # This just gets the docs stuff
+    response = requests.get("https://www.hikari-py.dev/" + "objects.inv")
+    cache = rtfm.decode_object_inv(response.content) #decodes it
+    plugin.d.hikari_cache = cache
+    response = requests.get("https://hikari-lightbulb.readthedocs.io/en/latest/" + "objects.inv")
+    cache = rtfm.decode_object_inv(response.content) #decodes it
+    plugin.d.lightbulb_cache = cache
```
Comment (vicky5124) [inline carberretta/extensions/meta.py:216]: This should be async, and be done somewhere else, such as `hikari.StartingEvent`

File: carberretta/extensions/meta.py [hunk 2]
```diff
@@ -40,9 +40,11 @@
 
 import carberretta
 from carberretta import Config
-from carberretta.utils import chron, helpers
+from carberretta.utils import chron, helpers, rtfm
 
-plugin = lightbulb.Plugin("Meta")
+import requests
```
Comment (vicky5124) [inline carberretta/extensions/meta.py:45]: You should not be using a blocking IO library with an async application. Replace this dependency with `aiohttp` instead.

File: carberretta/extensions/meta.py [hunk 3]
```diff
@@ -159,11 +161,59 @@ async def cmd_stats(ctx: lightbulb.SlashContext) -> None:
...
+@lightbulb.command("rtfm", description="Searches the docs of hikari and lightbulb.")
+@lightbulb.implements(lightbulb.SlashCommandGroup, lightbulb.PrefixCommandGroup)
+async def rtfm_group(ctx) -> None:
+    pass
+
+@rtfm.child
+@lightbulb.option("query", "The query to search for", autocomplete=True, required=True)
+@lightbulb.command("hikari", description="Searches the docs of hikari.", auto_defer=True)
+@lightbulb.implements(lightbulb.SlashSubCommand, lightbulb.PrefixSubCommand)
+async def hikari_rtfm(ctx):
```
Comment (vicky5124) [inline carberretta/extensions/meta.py:174]: Missing typehinting.

File: carberretta/extensions/meta.py [hunk 4]
```diff
@@ -159,11 +161,59 @@ async def cmd_stats(ctx: lightbulb.SlashContext) -> None:
         )
     )
 
+@plugin.command
+@lightbulb.command("rtfm", description="Searches the docs of hikari and lightbulb.")
+@lightbulb.implements(lightbulb.SlashCommandGroup, lightbulb.PrefixCommandGroup)
+async def rtfm_group(ctx) -> None:
```
Comment (vicky5124) [inline carberretta/extensions/meta.py:167]: Missing typehinting.

File: carberretta/extensions/meta.py [hunk 5]
```diff
@@ -159,11 +161,59 @@ async def cmd_stats(ctx: lightbulb.SlashContext) -> None:
...
+            embed.description += f"[`{match}`](https://www.hikari-py.dev/{plugin.d.hikari_cache[match][1]})\n"
+        except:
+            continue
+    await ctx.respond(embed=embed)
+
+@rtfm.child
+@lightbulb.option("query", "The query to search for", autocomplete=True, required=True)
+@lightbulb.command("lightbulb", description="Searches the docs of lightbulb.", auto_defer=True)
+@lightbulb.implements(lightbulb.SlashSubCommand, lightbulb.PrefixSubCommand)
+async def lightbulb_rtfm(ctx):
```
Comment (vicky5124) [inline carberretta/extensions/meta.py:189]: Missing typehinting.

File: carberretta/utils/rtfm.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
+# Hi carb
+
```
Comment (vicky5124) [inline carberretta/utils/rtfm.py:2]: This is not related to the code and should not be here.

Missing copyright notice.

File: carberretta/utils/rtfm.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+                chunk = decompressor.decompress(chunk) #decompresses to somethign
+                yield chunk #yields chunks
+            this = decompressor.flush()
+            yield this #yields this flush thing which is normally b''
+        cache = b""
+        for chunk in decomepress(bytes_obj):
+            cache += chunk
+            pos = cache.find(b"\n")
+            while pos != -1:
+                yield cache[:pos].decode("utf-8") #decodes to normal text
```
Comment (vicky5124) [inline carberretta/utils/rtfm.py:41]: This has not passed `black`.

**Review by vicky5124 — CHANGES_REQUESTED** [review]
In addition to all the small mistakes, the __pycache__ folder should not be part of the repository.

File: carberretta/utils/rtfm.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    if inv_version != "# Sphinx inventory version 2":
+        raise RuntimeError("Invalid object inv version")
+    name = bytes_obj.readline().decode("utf-8").rstrip()[11:] #This line is the project's name
+    version = bytes_obj.readline().decode("utf-8").rstrip()[11:] #This line is the project's version
+    zlib_line = bytes_obj.readline().decode("utf-8")
+    if "zlib" not in zlib_line: #The line that has Zlib in it
+        raise RuntimeError("Invalid object.inv file")
+    regex = re.compile(r"(?x)(.+?)\s+(\S*:\S*)\s+(-?\d+)\s+(\S+)\s+(.*)") #this thing is scary af
+    def decompress_chunks(bytes_obj):
+        def decomepress(bytes_obj):
```
Comment (Jonxslays) [inline carberretta/utils/rtfm.py:55]:
```suggestion
        def decompress(bytes_obj):
```
This goes for all the calls below as well.

**Review by Jonxslays — CHANGES_REQUESTED** [review]

**Comment by Jonxslays** [comment]
Theres tons of stuff that is untyped throughout. Everything needs to be typehinted.

**Comment by parafoxia** [comment]
@Coler6gamer, you'd benefit from doing `pip install -r requirements-dev.txt` if you haven't already, and running `nox`. It'll tell you what you need to change.

**Comment by IkBenOlie5** [comment]
That would also run the linting which would find things such as bare except on line 181 of meta.py.

**Comment by parafoxia** [comment]
I don't think so, unless Mypy does it or something. It's not explicitly set up to check that.

**Comment by IkBenOlie5** [comment]
Oh, I assumed you also used flake8 for linting just like hikari, but ig not.

**Comment by parafoxia** [comment]
Nah we only use it for imports atm.
